core/src/main/java/io/questdb/griffin/engine/functions/math/MulIntFunctionFactory.java

```diff
@@ -81,6 +81,17 @@ public Function getRight() {
             return right;
         }
 
+        @Override
+        public long getTimestamp(Record rec) {
```

Comment:
this should work generically as in: explicit constants should be treated the same way as implicit constants, e.g. `1720468802 * 1000000` and `1720468802000000` should be a long value.

Reply:
this should produce long value
```
select 1720468802 * 1000000;
```

Reply:
`+` and `-` should also be overflow aware